### PR TITLE
Describe how to add a new metric type using the macro

### PR DESCRIPTION
**Pull request by badboy**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: docs/dev/core/new-metric-type.md [hunk 1]
```diff
@@ -139,66 +139,43 @@ Each metric type is implemented in its own module.
 
 Add a new file named after your metric, e.g. `glean-core/ffi/src/counter.rs`, and declare it in `glean-core/ffi/src/lib.rs` with `mod counter;`.
 
-In the metric type module add a global map for your metric type and define the destructor:
+In the metric type module define your metric type using the `define_metric` macr.
```
Comment (mdboom) [inline docs/dev/core/new-metric-type.md]:
```suggestion
In the metric type module define your metric type using the `define_metric` macro.
```

File: docs/dev/core/new-metric-type.md [hunk 2]
```diff
@@ -139,66 +139,43 @@ Each metric type is implemented in its own module.
...
+This allows to reference the metric name and defines the global map as well as some common functions such as the constructor and destructor.
```
Comment (mdboom) [inline docs/dev/core/new-metric-type.md]:
```suggestion
This allows referencing the metric name and defines the global map as well as some common functions such as the constructor and destructor.
```

File: docs/dev/core/new-metric-type.md [hunk 3]
```diff
@@ -139,66 +139,43 @@ Each metric type is implemented in its own module.
...
-lazy_static! {
-    static ref COUNTER_METRICS: ConcurrentHandleMap<CounterMetric> = ConcurrentHandleMap::new();
-}
-define_handle_map_deleter!(COUNTER_METRICS, glean_destroy_counter_metric);
-```
+use crate::{define_metric, handlemap_ext::HandleMapExtension, GLEAN};
 
-Add a function to create new instances of this metric type:
+define_metric!(CounterMetric => COUNTER_METRICS {
+    new           -> glean_new_counter_metric(),
```
Comment (Dexterp37) [inline docs/dev/core/new-metric-type.md]: Not a problem but oh, I only noticed now that the `new` requires trailing `()`

Comment (badboy) [reply docs/dev/core/new-metric-type.md]: Only because it could also take additional parameters.
I _might_ be able to get rid of that, but not sure.

Comment (Dexterp37) [reply docs/dev/core/new-metric-type.md]: No worries, no need to act on this :)

**Pull request merged**